### PR TITLE
Quick Fix Summary issues

### PR DESCRIPTION
**Pull request by ash0080**
1. Calculate the string size without any html tags, so now you can get exactly what you wanted length;
2. Support utf8 2 character widths characters, like Chinese, Japanese;
3. This is a quick dirty mod, some associated functions should be rethinked;

### CODE REVIEW

**Comment by ash0080** [comment]
Compatible with the old version now

**Comment by Kasparsu** [comment]
`$this->home_route = $config->get('system.home.alias');` this was fine
`$this->home_route      = $config->get('system.home.alias');` this is not
if you do not know exactly what psr2 is then read up on it [http://www.php-fig.org/psr/psr-2/](here)
to ease your work you can install some linters for php aswell as. https://github.com/squizlabs/PHP_CodeSniffer and https://phpmd.org/ are useful. If you are using phpstorm you can easily set up code style and lint automatically from Code->Reformat code. Then you do not have to guess what is wanted from you and let the tools point it out.

File: system/src/Grav/Common/Page/Page.php [hunk 1]
```diff
@@ -106,16 +106,16 @@ public function __construct()
...
+        $this->taxonomy  = [];
+        $this->process   = $config->get('system.pages.process');
         $this->published = true;
     }
 
     /**
      * Initializes the page instance variables based on a file
      *
      * @param  \SplFileInfo $file The file information for the .md file that the page represents
-     * @param  string       $extension
```
Comment (Kasparsu) [inline system/src/Grav/Common/Page/Page.php]: why all the extra space? it was fine before. if you do not know exactly what psr2 is then read up on it [http://www.php-fig.org/psr/psr-2/](here) to ease your work you can install some linters for php aswell as. https://github.com/squizlabs/PHP_CodeSniffer and https://phpmd.org/ are useful. If you are using phpstorm you can easily set up code style and lint automatically from Code->Reformat code. Then you do not have to guess what is wanted from you and let the tools point it out.

File: system/src/Grav/Common/Page/Page.php [hunk 2]
```diff
@@ -215,17 +215,17 @@ public function translatedLanguages($onlyPublished = false)
      */
     public function untranslatedLanguages($includeUnpublished = false)
     {
-        $filename = substr($this->name, 0, -(strlen($this->extension())));
-        $config = Grav::instance()['config'];
-        $languages = $config->get('system.languages.supported', []);
+        $filename              = substr($this->name, 0, -(strlen($this->extension())));
```
Comment (Kasparsu) [inline system/src/Grav/Common/Page/Page.php]: why all the extra space? it was fine before. if you do not know exactly what psr2 is then read up on it [http://www.php-fig.org/psr/psr-2/](here) to ease your work you can install some linters for php aswell as. https://github.com/squizlabs/PHP_CodeSniffer and https://phpmd.org/ are useful. If you are using phpstorm you can easily set up code style and lint automatically from Code->Reformat code. Then you do not have to guess what is wanted from you and let the tools point it out.

Comment (ash0080) [inline system/src/Grav/Common/Page/Page.php]: Actually I use Reformat in PhpStorm，
that‘s the exact PSR-1&PSR-2 style predefined，
but seems not same as original code，
I will clean them manually for the next update

**Comment by rhukster** [comment]
Thanks so much!

**Comment by ash0080** [comment]
Thank you all！

**Pull request merged**